### ISSUE TITLE
Improve "double definition" of pi by an import statement

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1698]
**Reported by dietmarw on 16 Apr 2015 19:24 UTC**
Throughout the MSL (e.g., electrical machines, Spice3, fluid, mechanics) one can see the usage of:

```mo
 constant Modelica.SIunits.Angle pi=Modelica.Constants.pi;
```
or 
```mo
constant Real pi=Modelica.Constants.pi;
```

Especially the latter case might be problematic for unit derivation as some information of the type might get lost. 

I would suggest to replace those statements with the cleaner version of:

```mo
import pi = Modelica.Constants.pi;
```

unless I'm overlooking any drawback with this.

_There might be similar cases of other redefinitions which probably can be improved in the same way._

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1698

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 16 Apr 2015 19:26 UTC**
Actually make that:

```mo
import Modelica.Constants.pi;
```
;-)

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 11 Dec 2015 13:23 UTC**
Fixed in 33da3c980c83f023be7d36186a65ccbc6aa4aa97 as suggested.

Note, pi-definitions defined in partial models (that are inherited) are not replaced, because import is not inherited and suddenly models using the inherited class would miss the pi-definition.

**Comment by modelica-trac-importer** [comment]
**Changelog modified by otter on 11 Dec 2015 13:23 UTC**
Replaced "constant Real pi" definitions by import Modelica.Constants.pi declarations to avoid double definitions
